### ISSUE TITLE
Date: Option to change the default end date to the chosen start date

### ISSUE THREAD

**Issue by olafgrabienski** [backdrop/backdrop-issues#2752]
When you set up a date field to collect an end date in the field settings, you are asked for a default end date, with the options `Same as Default date`, `No default value`,  `Now`, and `Relative`. What I miss here, is the option `Same as start date`.

`Same as start date` is similar to `Same as Default date`, with the important difference that the latter is static; it leaves the end date at the Default date, no matter to what value the start date was changed by an editor). `Same as start date` in contrast would work dynamically: It would pick up the value of the start date chosen by an editor.

In my opinion, the `Same as start date` setting makes a lot of sense when you collect an end date and when an event is in the future, which is quite often the case. An example: You add an event node on first of July and choose first of October as the start date. The end date has to be by definition *after* the first of October. But with the end date setting `Same as Default date`, it defaults to July. So you first have to adjust the end date widget to jump from July to October (or later) before you can choose the end date. In contrast, with the `Same as start date` setting, the end date widget would switch to October based on the chosen start date, saving you some widget clicks.

**Comment by olafgrabienski** [comment]
Don't know how to label this issue: Feature request or [UX] bug report? @klonos, what do you think?

**Comment by Graham-72** [comment]
@olafgrabienski I support your request for this change as the present situation is causing me difficulty with the date_all_day contrib module which uses function date_is_all_day() from the date module to calculate whether the event lasts for more than one day, given the values of the from and to dates, I would prefer the default 'to' date to be set to be the same as the curremtly set 'from' date unless specifically set by the user. Issues https://github.com/backdrop-contrib/date_all_day/issues/5 and https://github.com/backdrop-contrib/date_all_day/issues/7 relate to this I think, and I am trying to devise a solution.

**Comment by olafgrabienski** [comment]
@Graham-72 Thanks for supporting my request! Are you also willing to work on this issue, or should we ask for help, or more feedback?

**Comment by Graham-72** [comment]
@olafgrabienski Yes, I am working on this issue at the moment and I hope soon to have a proposal for yourself and others to consider.

**Comment by olafgrabienski** [comment]
@Graham-72 (Didn't get a message about the PR, so I saw it only now by chance.) I've had a look at the PR on the sandbox site. Seems it didn't change the behaviour of the end date's default during content creation:

I added a *Date* field to the *Post* content type and checked if the *Default end date* was set to "Same as start date" which was the case. Then I added a new post and set the start date to 1st of october. When I put the cursor to the end date widget, the end date was not updated to october but remained at the current date (see screenshot).

![screen-pr-1923-default-end-date](https://user-images.githubusercontent.com/14188307/28251866-9ec4ada6-6a86-11e7-91ae-1be948957400.png)

**Comment by Graham-72** [comment]
@olafgrabienski yes, you are right. We have a problem. I will try and solve it.

**Comment by Graham-72** [comment]
@olafgrabienski it seems that this will work if the widget type for the date field is 'select list' but not if it is 'pop-up calendar'. This would seem to be a problem in the module (before applying this PR) where Javascript in `date.js` is used to make the end date match the start date. It should be possible to apply this more generally and not just for the 'select list' case.

**Comment by olafgrabienski** [comment]
@Graham-72 You're right, it works in the PR sandbox if the widget type is *Select list*. Good catch, I haven't checked the widget type at all! (There is also *Text field*, btw.)

Another observation regarding the PR: it works (with the Select list) if the *Default date* (i.e. the Default start date) is set to `No default value` but not if it's set to `Now`.

**Comment by Dinornis** [comment]
👍 for this feature request and labelling as such.

**Comment by nattywebdev** [comment]
Still a need for this valuable enhancement - see brief discussion on: 
https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Date.20field.20-.20end.20date.20to.20default.20to.20start.20date/near/444459292
